mantle/cmd/ore/aws/delete-image.go

```diff
@@ -57,6 +57,18 @@ func runDeleteImage(cmd *cobra.Command, args []string) {
 	}
 
 	if snapshotID != "" {
+		if snapshotID == "detectFromAMI" {
+			// Let's try to extract the snapshotID from AMI
+			snapshot, err := API.FindSnapshot(amiID)
+			// FindSnapshot returns nil if not found.
+			if err == nil {
```

Comment:
I think we need to check the `err != nil` case, right?

Reply:
The function `FindSnapshot` has multiple scenarios. Basically, its either snapshot is `nil` or `err!=nil` if nothing is found.

Reply:
the `FindSnapshot` function has several errors that it can throw:

- `unable to describe snapshots:`
- `found multiple matching snapshots`
- `unable to describe import tasks`
- `couldn't describe snapshot from import task`
- `found multiple matching import tasks`

in all of those cases you can't assume there is no snapshot. You can only say there is definitely no snapshot if `snapshot` and `err` are both `nil`.

Reply:
Understood, but they all mean we are not getting any snapshots from the AMI i.e. we won't be handling each scenario differently right?
From what I see, the scenario when we get a snapshot from AMI, the `snapshot!=nil` and `err==nil`. If anything else, it doesn't help us.

Reply:
It depends on how we want to handle it. If there was an AMI then there definitely is a snapshot and if we can't find a snapshot ID then we probably just want to error here. I mean we don't expect for an error to happen so bubbling that up to the user would be useful? or at least printing a warning (probably the better option).

Reply:
basically we should handle the `err != nil` case even if it's only informational:

```
if err != nil {
   print warning; do nothing
} else if snapshot == nil {
   print log message; do nothing
} else {
   is it possible to reach this?
}
```

Reply:
```
	if err != nil {
		fmt.Fprintf(os.Stderr, "Warning: Unable to find any snapshot for %s.\n", amiID)
		os.Exit(0)
	} else if snapshot == nil {
		fmt.Fprintf(os.Stderr, "No valid snapshot found for %s.\n", amiID)
		os.Exit(0)
	}
	snapshotID = snapshot.SnapshotID
```
Something like this  works?  We shouldn't need the `else` statement since that would just mean that we have a valid snapshot.

Reply:
that looks pretty good. I would actually say that we hit an error but we are going to continue. `fmt.Fprintf(os.Stderr, "Warning: Encountered error when searching for snapshot for %s: %s. Continuing..\n", amiID, err)`

also I wouldn't do `os.Stderr` for the non warning log message.